### PR TITLE
fix(misconf): do not filter Terraform plan JSON by name

### PR DESCRIPTION
**Pull request by nikpivkin**
## Description

Terraform Plan JSON is detected by content, so filtering files by name is unnecessary.

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/7393

### Before
```bash
ls
test.json

trivy conf .
2024-08-27T15:57:44+06:00	INFO	[misconfig] Misconfiguration scanning is enabled
2024-08-27T15:57:44+06:00	INFO	Detected config files	num=0
```

### After

```bash
ls
test.json trivy

./trivy conf .
2024/08/27 15:58:35 WARN Check ID is empty file_path=lib/kubernetes/utils.rego
2024/08/27 15:58:35 WARN Check ID is empty file_path=lib/docker/docker.rego
2024/08/27 15:58:35 WARN Check ID is empty file_path=lib/kubernetes/kubernetes.rego
2024-08-27T15:58:35+06:00	INFO	[misconfig] Misconfiguration scanning is enabled
2024-08-27T15:58:35+06:00	INFO	[terraform scanner] Scanning root modulefile_path="."
2024-08-27T15:58:35+06:00	INFO	Detected config files	num=2

main.tf (terraformplan)

Tests: 3 (SUCCESSES: 1, FAILURES: 2, EXCEPTIONS: 0)
Failures: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 2, CRITICAL: 0)

HIGH: Instance does not require IMDS access to require a token
════════════════════════════════════════════════════════════════════════════════

IMDS v2 (Instance Metadata Service) introduced session authentication tokens which improve security when talking to IMDS.
By default <code>aws_instance</code> resource sets IMDS session auth tokens to be optional.
To fully protect IMDS you need to enable session tokens by using <code>metadata_options</code> block and its <code>http_tokens</code> variable set to <code>required</code>.


See https://avd.aquasec.com/misconfig/avd-aws-0028
────────────────────────────────────────────────────────────────────────────────
 main.tf:8
   via main.tf:5-9 (metadata_options)
    via main.tf:1-10 (aws_instance.bad_example)
────────────────────────────────────────────────────────────────────────────────
   1   resource "aws_instance" "bad_example" {
   2   	ami = "ami-005e54dee72cc1d00"
   3   	instance_type = "t2.micro"
   4   	source_dest_check = true
   5   	metadata_options {
   6   	http_endpoint = "enabled"
   7   	http_protocol_ipv6 = "disabled"
   8 [ 	http_tokens = "optional"
   9   	}
  10   }
────────────────────────────────────────────────────────────────────────────────


HIGH: Root block device is not encrypted.
════════════════════════════════════════════════════════════════════════════════
Block devices should be encrypted to ensure sensitive data is held securely at rest.

See https://avd.aquasec.com/misconfig/avd-aws-0131
────────────────────────────────────────────────────────────────────────────────
 main.tf:1-10
────────────────────────────────────────────────────────────────────────────────
   1 ┌ resource "aws_instance" "bad_example" {
   2 │ 	ami = "ami-005e54dee72cc1d00"
   3 │ 	instance_type = "t2.micro"
   4 │ 	source_dest_check = true
   5 │ 	metadata_options {
   6 │ 	http_endpoint = "enabled"
   7 │ 	http_protocol_ipv6 = "disabled"
   8 │ 	http_tokens = "optional"
   9 │ 	}
  10 └ }
───────────────────────────────────────────────────────────────────────────────
```

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/iac/detection/detect.go [hunk 1]
```diff
@@ -150,8 +155,7 @@ func init() {
 			return false
 		}
 
-		return (sniff.Parameters != nil && len(sniff.Parameters) > 0) ||
-			(sniff.Resources != nil && len(sniff.Resources) > 0)
+		return len(sniff.Parameters) > 0 || len(sniff.Resources) > 0
```
Comment (simar7) [inline pkg/iac/detection/detect.go:158]: I believe we had the `nil` check to make sure we don't panic I case the fields are actually `nil`. Do we not need that?

**Comment by simar7** [comment]
Can we also add a test for this case?

File: pkg/iac/detection/detect.go (hunk 1 above)
Comment (nikpivkin) [reply pkg/iac/detection/detect.go:158]: The zero value of a slice is nil https://go.dev/tour/moretypes/12

**Comment by nikpivkin** [comment]
@simar7 Added a test case https://github.com/aquasecurity/trivy/pull/7406/commits/5b4ad737fab9a108896cf2ba583e463d5143ff9e

File: pkg/iac/detection/detect.go (hunk 1 above)
Comment (simar7) [reply pkg/iac/detection/detect.go:158]: ah good catch!

**Review by simar7 — APPROVED** [review]

**Pull request merged**